src/lib/client.js

```diff
@@ -26,18 +30,93 @@ module.exports = class BtpClient {
     this._wsUri = parsedBtpUri.toString().substring('btp+'.length)
   }
 
+  tryToOpenWebSocketClient () {
+    return new Promise((resolve, reject) => {
+      const ws = new WebSocket(this._wsUri, {
+        perMessageDeflate: false
+      })
+      console.log('try to open', this._wsUri)
+      let hasBeenOpen = false
+      ws.on('open', () => {
+        console.log('open!')
+        hasBeenOpen = true
+        resolve(ws)
+      })
+      ws.on('error', (err) => {
+        if (hasBeenOpen) {
```

Comment:
are you sure that the websocket can stay open after an error in this case?

Reply:
i'll check

Reply:
It seems indeed that an error event will always be followed by a close event. But the two cases addressed by the `if` there are different; in the first case, we want to log the error (even if we know a close event is on its way), and if the error comes before the ws has been open (e.g. the domain name doesn't resolve), then we want to reject the promise.

If we're absolutely sure that the close event always follows, we could move the reject call from the error handler to the close handler, but I'm not sure whether that would have any advantage?